Dependency on XStream, even when not using any XStream serializers

It's currently not possible to use Axon without XStream getting configured, even when XStream is not used for any serializers.

```
<dependency>
    <groupId>org.axonframework</groupId>
    <artifactId>axon-messaging</artifactId>
    <version>${version.axon}</version>
    <exclusions>
        <exclusion>
            <groupId>com.thoughtworks.xstream</groupId>
            <artifactId>xstream</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```

JpaSagaStore:361
> private Supplier<Serializer> serializer = XStreamSerializer::defaultSerializer;

> Caused by: java.lang.NoClassDefFoundError: com/thoughtworks/xstream/mapper/CannotResolveClassException
>    at deployment.war//org.axonframework.modelling.saga.repository.jpa.JpaSagaStore$Builder.<init>>(JpaSagaStore.java:361)

Same Issue was with QuartzEventScheduler :
Axon 4.5
[#1685] Remove QuartzEventScheduler hard dependency on XStream #1690
Remove hard dependency on XStream in QuartzEventScheduler #1685

--- a/modelling/src/main/java/org/axonframework/modelling/saga/repository/jpa/JpaSagaStore.java
+++ b/modelling/src/main/java/org/axonframework/modelling/saga/repository/jpa/JpaSagaStore.java
@@ -358,7 +358,7 @@ public class JpaSagaStore implements SagaStore<Object> {
     public static class Builder {
 
         private EntityManagerProvider entityManagerProvider;
-        private Supplier<Serializer> serializer = XStreamSerializer::defaultSerializer;
+        private Supplier<Serializer> serializer;
 
         /**
          * Sets the {@link EntityManagerProvider} which provides the {@link EntityManager} used to access the
@@ -404,6 +404,10 @@ public class JpaSagaStore implements SagaStore<Object> {
         protected void validate() throws AxonConfigurationException {
             assertNonNull(entityManagerProvider,
                           "The EntityManagerProvider is a hard requirement and should be provided");
+
+            if (serializer == null) {
+                serializer = XStreamSerializer::defaultSerializer;
+            }
         }
     }
 

